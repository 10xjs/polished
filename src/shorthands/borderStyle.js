// @flow
import directionalProperty from '../helpers/directionalProperty'

function borderStyle(...values: Array<string|null>) {
  return directionalProperty('border-style', ...values)
}

export default borderStyle
