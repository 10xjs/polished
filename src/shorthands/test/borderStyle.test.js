// @flow
import borderStyle from '../borderStyle'

describe('borderStyle', function() {
  it('properly applies a value when passed only one', function() {
    expect(borderStyle('12px')).toMatchSnapshot()
  })
  it('properly applies values when passed two', function() {
    expect(borderStyle('12px', '24px')).toMatchSnapshot()
  })
  it('properly applies values when passed three', function() {
    expect(borderStyle('12px', '24px', '36px')).toMatchSnapshot()
  })
  it('properly applies values when passed four', function() {
    expect(borderStyle('12px', '24px', '36px', '48px')).toMatchSnapshot()
  })
})
