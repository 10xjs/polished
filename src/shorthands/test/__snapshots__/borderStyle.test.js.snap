exports[`borderStyle properly applies a value when passed only one 1`] = `
Object {
  "border-bottom-style": "12px",
  "border-left-style": "12px",
  "border-right-style": "12px",
  "border-top-style": "12px",
}
`;

exports[`borderStyle properly applies values when passed four 1`] = `
Object {
  "border-bottom-style": "36px",
  "border-left-style": "48px",
  "border-right-style": "24px",
  "border-top-style": "12px",
}
`;

exports[`borderStyle properly applies values when passed three 1`] = `
Object {
  "border-bottom-style": "36px",
  "border-left-style": "24px",
  "border-right-style": "24px",
  "border-top-style": "12px",
}
`;

exports[`borderStyle properly applies values when passed two 1`] = `
Object {
  "border-bottom-style": "12px",
  "border-left-style": "24px",
  "border-right-style": "24px",
  "border-top-style": "12px",
}
`;
